core/src/executor/evaluate/stateless.rs

```diff
@@ -126,6 +126,14 @@ fn evaluate_function<'a>(
 
     match func {
         // --- text ---
+        Function::Concat(exprs) => {
+            let mut vec: Vec<Evaluated<'_>> = vec![];
+            for expr in exprs {
+                vec.push(eval(expr)?);
+            }
+
+            f::concat(name(), vec) // _s).await?),
+        }
```

Comment:
```suggestion
            let exprs = exprs.iter().map(eval).collect::<Result<_>>()?;

            f::concat(name(), exprs)
```

How about code like this?

Reply:
Awesome!  I was hoping to convert it so that it was more rust like.   I still don't understand it, but I guess in time I will. ;)